Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -3565,3 +3626,282 @@ def remove_tag_from_endpoints_command(client: CoreClient, args: Dict):
     return CommandResults(
         readable_output=f'Successfully removed tag {tag} from endpoint(s) {endpoint_ids}', raw_response=raw_response
     )
+
+
+def parse_risky_users_or_hosts(user_or_host: dict[str, Any], table_title: str) -> dict[str, Any]:
+    reasons = user_or_host.get('reasons', [])
+    return {
+        table_title: user_or_host.get('id'),
+        'Score': user_or_host.get('score'),
+        'Description': reasons[0].get('description') if reasons else None,
+    }
+
+
+def parse_user_groups(group: dict[str, Any]) -> list[dict[str, Any]]:
+    return [
+        {
+            'User email': user,
+            'Group Name': group.get('group_name'),
+            'Group Description': group.get('description'),
+        }
+        for user in group.get("user_email", [])
+    ]
+
+
+def parse_role_names(role_data: dict[str, Any]) -> dict[str, Any]:
+    return {
+        "Role Name": role_data.get("pretty_name"),
+        "Description": role_data.get("description"),
+        "Permission": role_data.get("permissions", []),
+        "Users": role_data.get("users", []),
+        "Groups": role_data.get("groups", []),
+    }
+
+
+def find_the_cause_error(err: Exception) -> str:
```

Comment:
```suggestion
def enrich_error_message_id_group_role(err: Exception) -> str | None:
```
Have it return either the enriched message, or `None` when not found